tests/js/spec/components/events/interfaces/frame/line.spec.tsx

```diff
@@ -0,0 +1,157 @@
+            ...data,
+          }}
+          registers={{}}
+          components={[]}
+          event={event}
+        />
+      );
+      expect(container).toSnapshot();
+    });
+  });
+
+  describe('renderContext()', () => {
+    it('should render context lines', () => {
+      const {queryByRole} = mountWithTheme(
+        <Line
+          data={{
+            ...data,
+            context: [
+              [
+                211,
+                '    # Mark the crashed thread and add its stacktrace to the exception',
+              ],
+              [212, "    crashed_thread = data['threads'][state.requesting_thread]"],
+              [213, "    crashed_thread['crashed'] = True"],
+            ],
+          }}
+          registers={{}}
+          components={[]}
+          event={event}
+          isExpanded
+        />
+      );
+      expect(queryByRole('list')).toSnapshot();
+    });
+
+    it('should render register values', () => {
+      const {getByText} = mountWithTheme(
+        <Line
+          data={data}
+          registers={{
+            r10: '0x00007fff9300bf70',
+            r11: '0xffffffffffffffff',
+            r12: '0x0000000000000000',
+            r13: '0x0000000000000000',
+            r14: '0x000000000000000a',
+            r15: '0x0000000000000000',
+            r8: '0x00007fff9300bf78',
+            r9: '0x0000000000000040',
+            rax: '0x00007fff9291e660',
+            rbp: '0x00007ffedfdff7e0',
+            rbx: '0x00007fff9291e660',
+            rcx: '0x0000000000000008',
+            rdi: '0x00007ffedfdff790',
+            rdx: '0x0000020000000303',
+            rip: '0x000000010fe00a59',
+            rsi: '0x0000000000000300',
+            rsp: '0x00007ffedfdff7c0',
+          }}
+          components={[]}
+          event={event}
+          isExpanded
+        />
+      );
+      expect(getByText('registers')).toBeInTheDocument();
+    });
+
+    it('should not render empty registers', () => {
+      const {queryByText} = mountWithTheme(
+        <Line data={data} registers={{}} components={[]} event={event} isExpanded />
+      );
+
+      expect(queryByText('registers')).toBeFalsy();
+    });
+
+    it('should render context vars', () => {
+      const vars = {
+        origin: null,
+        helper: '<sentry.coreapi.MinidumpApiHelper object at 0x10e157ed0>',
+        self: '<sentry.web.api.MinidumpView object at 0x10e157250>',
+        args: [],
+        request: '<WSGIRequest at 0x4531253712>',
+        content: '[Filtered]',
+        kwargs: {},
+        project_id: "u'3'",
+      };
+
+      const {getByText} = mountWithTheme(
+        <Line
+          data={{...data, vars}}
+          registers={{}}
+          components={[]}
+          event={event}
+          isExpanded
+        />
+      );
+
+      for (const [key, value] of Object.entries(vars)) {
+        const row = getByText(key).closest('tr');
+        expect(row).toBeTruthy();
+
+        if (!row) {
+          return;
```

Comment:
return or continue here?

Reply:
return... row can not be `null` as a `within` func parameter